### ISSUE TITLE
[Feature Request]: Provide an option to always draw debug geometry on top of everything else

### ISSUE THREAD

**Issue by visuallization** [DmitriySalnikov/godot_debug_draw_3d#7]
### Feature description

Hey there!

First of all, thanks for the great plugin. Having convenient debug drawing capabilities in godot with your plugin really makes life easier. 

I have a question/feature request: Is it possible to draw the debug geometry always on top of everything else. So e.g. if I would draw a line visuallizing a raycast, would it be possible to still see the line once it enters a mesh? I would also find this convenient e.g. when I am calculating some positions in a mesh and want to make sure I actually calculated the position correctly by displaying them with debug draws. Hope this makes sense.

Maybe this is already possible, but if not it would be great to have this option as well.

Cheers,
Florentin

### Solution approach

Provide an option to always draw debug geometry on top of everything else.

**Comment by DmitriySalnikov** [comment]
Do you mean to draw all the geometry on top of everything else? Or only certain `draw_*` calls, for example with the `no_depth_test=true` flag?
![image](https://user-images.githubusercontent.com/7782218/212463149-21ae76df-2ecb-45eb-b9fe-4087a00874ed.png)

If the first option, then it's not difficult to add (just change the material parameter), but if the second one, then I can use the idea from #3, like:
```gdscript
DebugDraw.set_depth_test(false)
DebugDraw.draw_box(...) # draw box on top of everything else
DebugDraw.set_depth_test(true)
DebugDraw.draw_sphere(...) # normal draw
```
And somewhere on `_init()`
```gdscript
DebugDraw.default_depth_test = true
```
or expose it in `ProjectSettings`.

And I suppose this is necessary for Godot 3 and 4?

**Comment by visuallization** [comment]
Hey there,
Thanks for the reply! 
Yes I was rather think about the second solution and support for Godot 3 + 4 would be really cool!

Regarding your proposed first solution, I think I still might be missing something. Can you assign a material to a debug draw call? Or do you mean I should set no_dept_test on the object I want to draw on, because if I do that, it does not seem to work.

Cheers

**Comment by DmitriySalnikov** [comment]
The screenshot was just for reference to the `No depth test` flag.
Now there is no way to change this parameter of the material through the `DebugDraw` API.

First I want to port what already worked in Godot 3, then I will add something new.

But for now, you can manually change this parameter of the materials
![Godot_v3 5 1-stable_win64_BK4xMg20bS](https://user-images.githubusercontent.com/7782218/212537807-4cd2d85c-4920-4227-b074-ef00630b2a08.gif)

**Comment by danbronsema** [comment]
This would be really helpful to be able to do in godot 4. Sometimes my points get hidden behind my objects and it'd be helpful to have them always appear on top.

**Comment by bassj** [comment]
I would also like this feature.

**Comment by kotx** [comment]
This would be great!

**Comment by DmitriySalnikov** [comment]
Since my gizmo is drawn with separate lines and arrows, disabling the depth test leads to incorrect sorting of objects.

https://github.com/DmitriySalnikov/godot_debug_draw_3d/assets/7782218/e1529b13-3fcc-492b-99ac-33ca1c2dc20b

But using wireframes it is not noticeable :)

https://github.com/DmitriySalnikov/godot_debug_draw_3d/assets/7782218/ee30269c-b89c-4070-936a-9c07b036b162

**Comment by DmitriySalnikov** [comment]
implemented in da6f67221a8bb879adac0409964cabcc91a12e40
